### PR TITLE
RequestedAuthnContext source selector

### PR DESCRIPTION
**Pull request by tvdijen**
Select the proper authsource based on the RequestedAuthnContext in an AuthnRequest.

### CODE REVIEW

**Comment by tvdijen** [comment]
To do:  I would like to implement the comparison != exact scenarios, by calling MutliAuth dynamically..

**Comment by tvdijen** [comment]
@thijskh Jaime tells me to just stick with 'exact' comparison and to run away from the rest.. What do you think? Would it be worth implementing?

File: modules/core/src/Auth/Source/AbstractSourceSelector.php [hunk 1]
```diff
@@ -58,14 +58,17 @@ public function __construct(array $info, array $config)
...
+            if ($as === null || !in_array($source[0], $this->validSources, true)) {
+                throw new Exception('Invalid authentication source: ' . $source[0]);
+            }
+
+            static::doAuthentication($as, $state);
+            return;
         }
-
-        static::doAuthentication($as, $state);
+        throw new Exception('Not implemented.');
```
Comment (thijskh) [inline modules/core/src/Auth/Source/AbstractSourceSelector.php:71]: Why are you going to the changes of making this return an array now if it's not implemented? Can you then just not leave it at the single value it was?

**Comment by thijskh** [comment]
I agree with Jaime. We had a similar discussion at SURFconext. We know of no known use of the other variants, it seems a theoretical thing from the spec, using exact is clear and sound what you will get. I would leave it out, it keeps the code a lot simpler and no one will miss the other variants.

File: modules/core/src/Auth/Source/AbstractSourceSelector.php (hunk 1 above)
Comment (tvdijen) [reply modules/core/src/Auth/Source/AbstractSourceSelector.php:71]: It was preparation for supporting the other comparisons.. Will revert

**Comment by tvdijen** [comment]
There appears to be something wrong with codecov... I noticed there wasn't an automatic comment in this PR, and now that I check on the codecov website I see only files in the src/ directory are covered and the modules/ directory seems ignored

**Review by thijskh — APPROVED** [review]

**Pull request merged**